packages/plugins/users-permissions/server/controllers/auth.js

```diff
@@ -339,6 +339,16 @@ module.exports = {
       throw new ApplicationError('Email is already taken');
     }
 
+    if (params.username) {
```

Comment:
The only concern I have reviewing the PR is that since the username is an optional param here it won't prevent registrations without a username anyway and can be worked around by any user :/

Reply:
What do you think about it @WalkingPizza ?

Reply:
@alexandrebodin, what you are saying does make a lot of sense, what do we do?
Maybe we could consider offering users a way to extend U&P to make usernames mandatory and unique, and potentially also tackle what I discussed with @petersg83 in #12301.

Reply:
Actually the username is required via the model validation it self so this could work. I'm going to dig a bit more before moving forward :)